pac4j-core/src/main/java/org/pac4j/core/engine/DefaultSecurityLogic.java

```diff
@@ -145,7 +149,9 @@ public R perform(final C context, final Config config, final SecurityGrantedAcce
                 } else {
                     if (startAuthentication(context, currentClients)) {
                         logger.debug("Starting authentication");
-                        saveRequestedUrl(context, currentClients);
+                        if (!ajaxRequestResolver.isAjax(context)) {
```

Comment:
I'd prefer to have this logic `if (!ajaxRequestResolver.isAjax(context)) {` inside the `saveRequestedUrl` to make this overridable.